android/src/main/kotlin/com/almoullim/background_location/BackgroundLocationService.kt

```diff
@@ -77,8 +78,13 @@ class BackgroundLocationService: MethodChannel.MethodCallHandler, PluginRegistry
 
         receiver = MyReceiver()
```

Comment:
If the receiver is only ever registered if it had not been registered before, then we should also not re-reinitialize it. Simply put: Why is this not part of the following `if` block?

And more generally, the code at the beginning of this method should also be reviewed: Should that also maybe be not run if the receiver had already been registered?